home-excercises-05-2022/src/io/codelex/arithmetic/practice/CheckOddEven.java

```diff
@@ -0,0 +1,17 @@
+package io.codelex.arithmetic.practice;
+
+import java.util.Scanner;
+
+public class CheckOddEven {
+    public static void main(String[] args) {
+        System.out.println("Please enter Your integer number:");
+        Scanner input = new Scanner(System.in);
+        int num = input.nextInt();
+        if ((num & 1) == 1) {
```

Comment:
Vai zini, kā šis patiesībā strādā? :) Un ko tieši dara & operators? Diezgan sarežģīts piegājiens priekš tā, lai saprastu vai skaitlis ir pāra/nepāra. :D

Reply:
Sveiks! Es domāju, ka zinu kā strādā algoritms un & binārais salīdzināšanas operators. Tā ir binārā salīdzināšana un pāra skaitļiem vienmēr ir nulle binārā veidā galā, bet nepāra vieninieks - bet vieninieks bināri ir nulles un 1 galā un ar & bināri summējot līdz ar to vienmēr sākums rezutlātam vienmēr būs nulles un pēdējais cipars 0 ja pāra, bet 1 ja nepāra skaitlis. Varēja taisīt ar modulo, bet šis šķita interesantāks variants :)

Reply:
Jebkurā gadījumā es šo pieņemu kā risinājumu, bet ņem vērā, ka vienmēr vienkāršākais risinājums ir labākais. Sarežģītam risinājumam ir noteikti jābūt arī tehniski pamatotam. Šinī gadījumā tikai mana pieredze saka priekšā, ka 99% gadījumu šī rindiņa liktu programmētājiem smagi padomāt, lai saprastu ko tā dara. :D Jo bitwise operatori tiek izmantoti ļoooti reti.

Reply:
🙏